File: typescript/lib/core/graph/node.ts
export type NodeNeighbor = [string, number];

export class Node {
  constructor(
    private readonly _name: string,
    private readonly _neighbors: NodeNeighbor[],
  ) {}

  get name(): string {
    return this._name;
  }

  get neighbors(): NodeNeighbor[] {
    return this._neighbors;
  }
}

File: python/lib/core/graph/node.py
from typing import List, Tuple

NodeNeighbor = Tuple[str, int]


class Node:
    def __init__(self, name: str, neighbors: List[NodeNeighbor]) -> None:
        self._name = name
        self._neighbors = neighbors

    @property
    def name(self) -> str:
        return self._name

    @property
    def neighbors(self) -> List[NodeNeighbor]:
        return self._neighbors
